### ISSUE TITLE
wkhtmltopdf export pdf to html with js under IIS

### ISSUE THREAD

**Issue by Oni974** [wkhtmltopdf/wkhtmltopdf#1581]
Hello everyone (sorry for my bad english...),

I do not post here often, because I generally find my solutions on the web.

But then I dries clear.

In WAMP everything works perfectly. By cons on the IIS 7.5 server, I can not export my pdf via the command:

"C: \ wkhtmltopdf \ bin \ wkhtmltopdf.exe" D: \ app \ temp \ tmp77D9.tmp.html - enable-javascript D: \ app \ temp \ tmp77DA.tmp

For all pages with no javascript, it works impeccable. But as soon as I put the script tags that no longer works.

To summarize:

Full page with js appears correctly on the browser as WAMP and IIS (via vpn).

Full page with js runs correctly using the command on wkhtmltopdf in wamp (I have my pdf) but not on IIS (I have a pdf 0ko).

Without the js page runs correctly using the command on wkhtmltopdf under wamp and IIS.

If you want more details thank you for telling me.

**Comment by AstronautFarmer** [comment]
Hello,
I think there was misunderstanding. @Oni974 said that converting goes fine when it is run under wamp and IIS and html doesn't contain js, so it means that there is no issue with permission/setup on IIS. This issue appears when html contains js and only on IIS, under wamp it goes fine even with js. I know it because I had the same issue. I've tried use --disable-javascript but it didn't help. Also process returns Exit Code = -1073741819. But seems it is not because of images, process makes converting without error if html doesn't contain js but contains the same images. 
I have IIS 8 on Windows 8 x64 and use 0.12.2.0 x64 version.

**Comment by ashkulz** [comment]
If it works from the command-line but not from any other environment, then I cannot help you with that unless you can **prove** it is an issue with `wkhtmltopdf`.

**Comment by ramanmittal-kiwi** [comment]
I had the same problem
What helped me is this: Setting in Application Pool Identity = LocalSystem

Open Application Pools
Find your pool (by default DefaultAppPool)
Click on "Advanced settings"
Change Identity to "LocalSystem

**Comment by frankleenDev** [comment]
Thank you for the quick reply. 

Too bad I’m running the system on a shared hosting environment at godaddy of which I can’t access the option since it’s only available to dedicated server clients.
I also tried to change the wkhtmltopdf.exe to a wkhtmltopdf from Linux, how ever it required libraries I couldn’t provide.
Any idea friend. 

Sent from Mail for Windows 10
